### PR TITLE
feat: Export TypeScript types

### PR DESCRIPTION
**Pull request by nwalters512**
**🤔Type of Request**
- [ ] **Bug fix**
- [ ] **New feature**
- [x] **Improvement**
- [ ] **Documentation**
- [ ] **Other**

**📝Changelog**

- [x] **Core**
- [ ] **Extensions**

This PR exports some of the types from `index.d.ts`. This improves the ability of this library to be used with TypeScript, e.g.

```ts
import { BootstrapTableOptions } from 'bootstrap-table';

const options: BootstrapTableOptions = {
  // ...
};
```

Importantly, this also allows us to use module augmentation to e.g. add extra properties to the `BootstrapTableColumn` type if we're using extensions.

**☑️Self Check before Merge**

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -1,22 +1,6 @@
 /// <reference types="jquery" />
 
-interface BootstrapTableHtml{
```
Comment (nwalters512) [inline index.d.ts:3]: This interface was unused, so I removed it.

File: index.d.ts [hunk 2]
```diff
@@ -301,8 +285,10 @@ interface BootstrapTableOptions{
     pageList?: number[];
 }
 
-interface JQuery{
-    bootstrapTable(options: BootstrapTableOptions): JQuery;
+declare global {
```
Comment (nwalters512) [inline index.d.ts:288]: This was necessary now that this file is considered a module.

**Comment by wenzhixin** [comment]
This branch has conflicts that must be resolved

**Review by wenzhixin — APPROVED** [review]

**Comment by wenzhixin** [comment]
@nwalters512

**Comment by nwalters512** [comment]
@wenzhixin apologies for the delay, the conflicts are resolved now!

**Review by wenzhixin — APPROVED** [review]

**Pull request merged**